### PR TITLE
Feature/dray/test class coverage

### PR DESCRIPTION
**Pull request by prescod**
# Changes

The `run_tasks` task  now has a `required_per_class_code_coverage_percent` option which ensures that every class in your project meets the code coverage level specifed.

----

Code strings to be reviewed are:
        self.logger.info("Checking code coverage.")

f"All classes meet code coverage expectations of XX% ."

f"Organization-wide code coverage of XX% is below required level of XX"

f"Organization-wide code coverage of XX% meets expectations."

f"{class_name}'s code coverage of XX% is below required level of YY."

Some improvements to the integration testing framework were also included.

### CODE REVIEW

File: cumulusci.yml [hunk 1]
```diff
@@ -63,6 +68,27 @@ flows:
...
+        description: Test the run_tests task in a CumulusCI-Test org
+        steps:
+            1:
+                flow: ccitest:qa_org
+
+            2:
+                task: run_tests
+                options:
+                    required_per_class_code_coverage_percent: 50
+            3:
```
Comment (jofsky) [inline cumulusci.yml:81]: @prescod  why run this twice with the same parameters?

File: cumulusci/tasks/apex/tests/test_apex_tasks.py [hunk 2]
```diff
@@ -1280,3 +1283,65 @@ def test_job_not_found(self):
             task()
 
         assert "found" in str(e.exception)
+
+
+class TestApexIntegrationTests:
+    @pytest.mark.org_shape("qa", "ccitest:qa_org")
+    @pytest.mark.slow()
```
Comment (jofsky) [inline cumulusci/tasks/apex/tests/test_apex_tasks.py:1290]: what's this do?

File: cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py [hunk 3]
```diff
@@ -113,7 +114,12 @@ def org_config(request, current_org_shape, cli_org_config, fallback_org_config):
     # fast running test suites it might return a hardcoded
     # org and for integration test suites it might return
     # a specific default org or throw an exception.
-    return org_config
+    with mock.patch.object(OrgConfig, "latest_api_version", "48.0"), mock.patch.object(
+        OrgConfig, "refresh_oauth_token"
+    ):
+        print("XXX yieldiing orgconfig", OrgConfig.refresh_oauth_token)
```
Comment (jofsky) [inline cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py:120]: should we log and not print this to maintain CumulusCI's use of logger?

**Review by jofsky — DISMISSED** [review]

File: cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py (hunk 3 above)
Comment (prescod) [reply cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py]: Fixed in 3ba5ceedafe which I guess I forgot to push.

**Review by prescod — COMMENTED** [review]
Test.

**Review by prescod — COMMENTED** [review]
Test.

**Review by prescod — COMMENTED** [review]
Test.

**Review by davidjray — APPROVED** [review]
Reviewed latest commits, and it looks good to me.

File: cumulusci/tasks/apex/testrunner.py [hunk 4]
```diff
@@ -507,10 +515,11 @@ def _init_task(self):
             )
 
     def _run_task(self):
+
         result = self._get_test_classes()
         if result["totalSize"] == 0:
             return
-        for test_class in result["records"]:
+        for test_class in result["records"][-20:]:
```
Comment (davisagli) [inline cumulusci/tasks/apex/testrunner.py:522]: @prescod Why did this change to only run the last 20 test classes?

**Pull request merged**